### ISSUE TITLE
Feature: (sort-switch-case) Would you consider sorting the `case` blocks in a `switch` statement?

### ISSUE THREAD

**Issue by u3u** [azat-io/eslint-plugin-perfectionist#54]
### Describe the rule

Can you add a `sort-switch-case` rule? It will sort the `case` statements, their order does not affect the execution result, but it can improve readability, reduce code merge conflicts, and eliminate the need to worry about their order when writing code.

### Code example

**Input**

```ts
switch (type) {
  case 'save': {
    break;
  }

  case 'edit': {
    break;
  }

  default:
  case 'close': {
    break;
  }
}
```

**Output**

```ts
switch (type) {
  default:
  case 'close': {
    break;
  }

  case 'edit': {
    break;
  }

  case 'save': {
    break;
  }
}
```

### Validations

- [X] Read the [docs](https://eslint-plugin-perfectionist.azat.io).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
That's a cool idea. I think we should add that.

**Comment by Pyrolistical** [comment]
The form `switch(true)` will become more popular with the release of typescript 5.3 https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-3.html#switch-true-narrowing

That also makes this feature more difficult to implement if switching the order of cases causes type errors, but maybe that is just revealing a bug

**Comment by azat-io** [comment]
Thank you for your issue!

It was a great idea. If you have any more suggestions for improving the plugin, feel free to create a new issue.

This feature was implemented in 5311118ebea707e0b885c4cf6ecbd84d10ea08d2 and released in [v3.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v3.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1815367279191761054

**Comment by u3u** [comment]
<img width="320" alt="image" src="https://github.com/user-attachments/assets/2ad0e9ff-c1e9-4d97-8e30-d8eaab2cd7c5">
<img width="200" alt="image" src="https://github.com/user-attachments/assets/a17f4953-4ce0-475c-9478-852b5d5ddea7">
<img width="300" alt="image" src="https://github.com/user-attachments/assets/d132be48-817e-4fa8-8af3-330aef89578b">
<img width="320" alt="image" src="https://github.com/user-attachments/assets/053157de-18a9-4a57-a87b-b89ec84d78f1">

@azat-io Hi, thank you for implementing this!! but it seems that it cannot be used together with [`unicorn/switch-case-braces`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v55.0.0/docs/rules/switch-case-braces.md). And there is a slight issue with sorting when `case` and `default` are used together.
